symphony/bdk/core/symphony_bdk.py

```diff
@@ -26,6 +37,26 @@ async def bot_session(self) -> AuthSession:
             self._bot_session = await self._authenticator_factory.get_bot_authenticator().authenticate_bot()
         return self._bot_session
 
+    async def messages(self):
```

Comment:
is it ok to make it async?

For instance should we perform any remote call when creating the message service?
Or should we do the first calls only when actually using the service for the first time?

Reply:
the problem here is we need a bot session to initialise the message service instance. So I guess, we should pass the `authenticator` to the message service constructor then inside the message service, each time we make a api call we gonna check the bot session there?

Reply:
I think it's okay to leave this method async. In fact, it's only really async at the first time we authenticate the bot to get the `bot_session`.
We can also init the bot session as `AuthSession(bot_authenticator)`, but then inside the message, at the beginning of each service method, we have to check if the token is None or not, then authenticate the bot, and TBH I don't really like this solution.
WDYT?

Reply:
What if the check is part of the getter in AuthSession?
One problem I see is that in case of concurrent access we might authenticate twice (if we don't lock) but with async it is not going to happen anyway